sdk/instrumentation/instrumentation-opentelemetry/src/instrumentation.ts

```diff
@@ -26,24 +26,25 @@ class AzureSDKInstrumentation extends InstrumentationBase {
   }
   /**
    * Entrypoint for the module registration.
-   * 
+   *
    * @returns The patched \@azure/core-tracing module after setting its instrumenter.
    */
   protected init():
     | void
     | InstrumentationModuleDefinition<typeof coreTracing>
     | InstrumentationModuleDefinition<typeof coreTracing>[] {
-    const result: InstrumentationModuleDefinition<typeof coreTracing> = new InstrumentationNodeModuleDefinition(
-      "@azure/core-tracing",
-      ["^1.0.0-preview.14", "^1.0.0"],
-      (moduleExports) => {
-        if (typeof moduleExports.useInstrumenter === "function") {
-          moduleExports.useInstrumenter(new OpenTelemetryInstrumenter());
-        }
+    const result: InstrumentationModuleDefinition<typeof coreTracing> =
+      new InstrumentationNodeModuleDefinition(
+        "@azure/core-tracing",
+        ["^1.0.0-preview.14", "^1.0.0"],
+        (moduleExports) => {
+          if (typeof moduleExports.useInstrumenter === "function") {
+            moduleExports.useInstrumenter(new OpenTelemetryInstrumenter());
```

Comment:
should we replace this class with a factory too?

Reply:
Unfortunately we have to inherit from their `InstrumentationBase` and override `init` - we only expose the factory though